contracts/EthlanceJobs.sol

```diff
@@ -0,0 +1,843 @@
+
+    emit JobFulfilled(_jobId,
+                      (jobs[_jobId].fulfillments.length - 1),
+                      _fulfillers,
+                      _ipfsHash, // The _ipfsHash string is emitted in an event for easy off-chain consumption
+                      _sender,
+                      _amount);
+  }
+
+  /// @dev updateFulfillment(): Allows the submitter of a fulfillment to update their submission
+  /// @param _sender the sender of the transaction issuing the job (should be the same as msg.sender unless the txn is called by the meta tx relayer)
+  /// @param _jobId the index of the job
+  /// @param _fulfillmentId the index of the fulfillment
+  /// @param _fulfillers the new array of addresses which will receive payouts for the submission
+  /// @param _ipfsHash the new IPFS hash corresponding to a JSON object which contains the details of the submission (see docs for schema details)
+  function updateFulfillment(
+                             address _sender,
+                             uint _jobId,
+                             uint _fulfillmentId,
+                             address payable[] memory _fulfillers,
+                             string memory _ipfsHash)
+    public
+    senderIsValid(_sender)
+    validateJobArrayIndex(_jobId)
+    validateFulfillmentArrayIndex(_jobId, _fulfillmentId)
+    onlySubmitter(_sender, _jobId, _fulfillmentId) // Only the original submitter of a fulfillment may update their submission
+  {
+    require(contains(jobs[_jobId].hiredCandidates, _sender));
+    jobs[_jobId].fulfillments[_fulfillmentId].fulfillers = _fulfillers;
+    emit FulfillmentUpdated(_jobId,
+                            _fulfillmentId,
+                            _fulfillers,
+                            _ipfsHash); // The _ipfsHash string is emitted in an event for easy off-chain consumption
+  }
+
+  /// @dev acceptFulfillment(): Allows any of the approvers to accept a given submission
+  /// @param _sender the sender of the transaction issuing the job (should be the same as msg.sender unless the txn is called by the meta tx relayer)
+  /// @param _jobId the index of the job
+  /// @param _fulfillmentId the index of the fulfillment to be accepted
+  /// @param _approverId the index of the approver which is making the call
+  /// @param _tokenAmounts the array of token amounts which will be paid to the
+  ///                      fulfillers, whose length should equal the length of the
+  ///                      _fulfillers array of the submission. If the job pays
+  ///                      in ERC721 tokens, then these should be the token IDs
+  ///                      being sent to each of the individual fulfillers
+  function acceptFulfillment(
+                             address _sender,
+                             uint _jobId,
+                             uint _fulfillmentId,
+                             uint _approverId,
+                             uint[] memory _tokenAmounts)
```

Comment:
acceptFulfillment can be called by approvers and can specify _tokenAmounts. I suppose in original StandardBounties, issuer was added as an approver also. But when issuer accepts fulfilment, he shouldn't be able to specify amount. He has to pay amount the amount fulfiller specified when calling fulfillJob. And when approver (arbiter) is resolving dispute, he should be able to specify the amount.

Reply:
the amount in fulfillJob is the total amount but it only indicates how much to each fulfiller inside meta in ipfs. The array here (_tokenAmounts) is the distribution to fulfillers. What we can check maybe is that the sum(_tokenAmounts) == fulfillment amount

Reply:
I think best would be if we remove feature of multiple fulfillers altogether. I can't think of proper use case for that on Ethlance.